metal/config.go

```diff
@@ -49,14 +51,13 @@ func MetalRetryPolicy(ctx context.Context, resp *http.Response, err error) (bool
 
 // Client returns a new client for accessing Equinix Metal's API.
 func (c *Config) Client() *packngo.Client {
-	httpClient := retryablehttp.NewClient()
-	httpClient.RetryWaitMin = time.Second
-	httpClient.RetryWaitMax = 30 * time.Second
-	httpClient.RetryMax = 10
-	httpClient.CheckRetry = MetalRetryPolicy
-	httpClient.HTTPClient.Transport = logging.NewTransport(
-		"Equinix Metal",
-		httpClient.HTTPClient.Transport)
-
+	transport := logging.NewTransport("Equinix Metal", http.DefaultTransport)
+	retryClient := retryablehttp.NewClient()
+	retryClient.HTTPClient.Transport = transport
```

Comment:
I don't think this will work because of what `.StandardClient()` does:
https://github.com/hashicorp/go-retryablehttp/blob/991b9d0a42d13014e3689dd49a94c02be01f4237/client.go#L772

What do you think about this: https://github.com/equinix/terraform-provider-metal/pull/21#discussion_r568004813

Reply:
It works, the RoundTripper wraps the retryable client which has it's own http.Client which still keeps the transport:
```
	transport := logging.NewTransport("Equinix Metal", http.DefaultTransport)
	retryClient := retryablehttp.NewClient()
	retryClient.HTTPClient.Transport = transport
        [...]
	httpClient := retryClient.StandardClient()
	t := httpClient.Transport
	rhrt := t.(*retryablehttp.RoundTripper)
	fmt.Println("EEEEEEEEEEEEEEEEEE", rhrt.Client.HTTPClient.Transport)
```

.. prints `EEEEEEEEEEEEEEEEEE &{Equinix Metal 0x1b2f4e0}`

> What do you think about this: #21 (comment)
> `httpClient := retryClient.StandardClient()`
> `httpClient.HTTPClient.Transport = transport`

That wouldn't compile.